seqfold/dna.go

```diff
@@ -0,0 +1,519 @@
+	"TT/AC": {1.0, 0.7},
+	"AA/TC": {2.3, 4.6},
+	"AC/TA": {5.3, 14.6},
+	"CA/GC": {1.9, 3.7},
+	"CC/GA": {0.6, -0.6},
+	"GA/CC": {5.2, 14.2},
+	"GC/CA": {-0.7, -3.8},
+	"TA/AC": {3.4, 8.0},
+	"TC/AA": {7.6, 20.2},
+	"AA/TA": {1.2, 1.7},
+	"CA/GA": {-0.9, -4.2},
+	"GA/CA": {-2.9, -9.8},
+	"TA/AA": {4.7, 12.9},
+	"AC/TC": {0.0, -4.4},
+	"CC/GC": {-1.5, -7.2},
+	"GC/CC": {3.6, 8.9},
+	"TC/AC": {6.1, 16.4},
+	"AG/TG": {-3.1, -9.5},
+	"CG/GG": {-4.9, -15.3},
+	"GG/CG": {-6.0, -15.8},
+	"TG/AG": {1.6, 3.6},
+	"AT/TT": {-2.7, -10.8},
+	"CT/GT": {-5.0, -15.8},
+	"GT/CT": {-2.2, -8.4},
+	"TT/AT": {0.2, -1.5},
+}
+
+// Terminal mismatch table (DNA)
+// SantaLucia & Peyret (2001) Patent Application WO 01/94611
+var DNA_TERMINAL_MM = BpEnergy{
+	"AA/TA": {-3.1, -7.8},
+	"TA/AA": {-2.5, -6.3},
+	"CA/GA": {-4.3, -10.7},
+	"GA/CA": {-8.0, -22.5},
+	"AC/TC": {-0.1, 0.5},
+	"TC/AC": {-0.7, -1.3},
+	"CC/GC": {-2.1, -5.1},
+	"GC/CC": {-3.9, -10.6},
+	"AG/TG": {-1.1, -2.1},
+	"TG/AG": {-1.1, -2.7},
+	"CG/GG": {-3.8, -9.5},
+	"GG/CG": {-0.7, -19.2},
+	"AT/TT": {-2.4, -6.5},
+	"TT/AT": {-3.2, -8.9},
+	"CT/GT": {-6.1, -16.9},
+	"GT/CT": {-7.4, -21.2},
+	"AA/TC": {-1.6, -4.0},
+	"AC/TA": {-1.8, -3.8},
+	"CA/GC": {-2.6, -5.9},
+	"CC/GA": {-2.7, -6.0},
+	"GA/CC": {-5.0, -13.8},
+	"GC/CA": {-3.2, -7.1},
+	"TA/AC": {-2.3, -5.9},
+	"TC/AA": {-2.7, -7.0},
+	"AC/TT": {-0.9, -1.7},
+	"AT/TC": {-2.3, -6.3},
+	"CC/GT": {-3.2, -8.0},
+	"CT/GC": {-3.9, -10.6},
+	"GC/CT": {-4.9, -13.5},
+	"GT/CC": {-3.0, -7.8},
+	"TC/AT": {-2.5, -6.3},
+	"TT/AC": {-0.7, -1.2},
+	"AA/TG": {-1.9, -4.4},
+	"AG/TA": {-2.5, -5.9},
+	"CA/GG": {-3.9, -9.6},
+	"CG/GA": {-6.0, -15.5},
+	"GA/CG": {-4.3, -11.1},
+	"GG/CA": {-4.6, -11.4},
+	"TA/AG": {-2.0, -4.7},
+	"TG/AA": {-2.4, -5.8},
+	"AG/TT": {-3.2, -8.7},
+	"AT/TG": {-3.5, -9.4},
+	"CG/GT": {-3.8, -9.0},
+	"CT/GG": {-6.6, -18.7},
+	"GG/CT": {-5.7, -15.9},
+	"GT/CG": {-5.9, -16.1},
+	"TG/AT": {-3.9, -10.5},
+	"TT/AG": {-3.6, -9.8},
+}
+
+//DNA dangling ends
+//
+//Bommarito et al. (2000), Nucl Acids Res 28: 1929-1934
+var DNA_DE = BpEnergy{
+	"AA/.T": {0.2, 2.3},
+	"AC/.G": {-6.3, -17.1},
+	"AG/.C": {-3.7, -10.0},
+	"AT/.A": {-2.9, -7.6},
+	"CA/.T": {0.6, 3.3},
+	"CC/.G": {-4.4, -12.6},
+	"CG/.C": {-4.0, -11.9},
+	"CT/.A": {-4.1, -13.0},
+	"GA/.T": {-1.1, -1.6},
+	"GC/.G": {-5.1, -14.0},
+	"GG/.C": {-3.9, -10.9},
+	"GT/.A": {-4.2, -15.0},
+	"TA/.T": {-6.9, -20.0},
+	"TC/.G": {-4.0, -10.9},
+	"TG/.C": {-4.9, -13.8},
+	"TT/.A": {-0.2, -0.5},
+	".A/AT": {-0.7, -0.8},
+	".C/AG": {-2.1, -3.9},
+	".G/AC": {-5.9, -16.5},
+	".T/AA": {-0.5, -1.1},
+	".A/CT": {4.4, 14.9},
+	".C/CG": {-0.2, -0.1},
+	".G/CC": {-2.6, -7.4},
+	".T/CA": {4.7, 14.2},
+	".A/GT": {-1.6, -3.6},
+	".C/GG": {-3.9, -11.2},
+	".G/GC": {-3.2, -10.4},
+	".T/GA": {-4.1, -13.1},
+	".A/TT": {2.9, 10.4},
+	".C/TG": {-4.4, -13.1},
+	".G/TC": {-5.2, -15.0},
+	".T/TA": {-3.8, -12.6},
+}
+
+// the energies are the same for each loop stack in the
+// reverse complementary direction
+func init() {
```

Comment:
what is this initializing?

Reply:
It is adding the reverse of each sequence in the the energy maps, since they have the same energy IIUC.
Also I left the python code commented out as reference to be sure I was doing the right thing.

Reply:
Unrolled the maps and removed the `init()`